google/cloud/storage/_http.py

```diff
@@ -46,3 +48,15 @@ def __init__(self, client, client_info=None, api_endpoint=DEFAULT_API_ENDPOINT):
 
     API_URL_TEMPLATE = "{api_base_url}/storage/{api_version}{path}"
     """A template for the URL of a particular API call."""
+
+    def api_request(self, *args, retry=None, **kwargs):
+        call = functools.partial(super(Connection, self).api_request, *args, **kwargs)
+        if retry:
+            # If this is a ConditionalRetryPolicy, check conditions.
+            try:
+                retry = retry.get_retry_policy_if_conditions_met(**kwargs)
+            except AttributeError:
```

Comment:
Duck typing: if we get an AttributeError here it is because the retry policy passed in was a straight run-of-the-mill retry policy, and not a ConditionalRetryPolicy. We could just do an isinstance check, but duck typing is usually preferred in Python. I'll document this in a comment.